consumer/consumer.py

```diff
@@ -0,0 +1,345 @@
+        self._minio_client = minio_client
+
+    def get_file_size(self, file_path: str) -> int:
+        object_stat = self._minio_client.stat_object(self._root_path, file_path)
+        return object_stat.size
+
+    def get_full_path(self):
+        date_dict = self.parse_received_at_date()
+
+        # (1) Define folder path
+        folder_path = os.path.join(
+            date_dict["year"], date_dict["month"], date_dict["day"], date_dict["hour"]
+        )
+        # (2) Get all json files in the dir
+        objects = self._minio_client.list_objects(self._root_path, prefix=folder_path, recursive=False)
+        json_files = [obj.object_name for obj in objects if obj.object_name.endswith('.json')]
+
+        # (3) Check existing files
+        file_path = self.get_or_create_file_path(json_files, folder_path, date_dict)
+
+        return file_path
+
+    def write_file(self):
+        # (1) Get path
+        file_path = self.get_full_path()
+
+        # (2) Write
+        # (2.1) Read existing data if the file exists
+        try:
+            response = self._minio_client.get_object(self._root_path, file_path)
+            existing_data = json.load(response)
+            response.close()
+            response.release_conn()  # Important to close the response to avoid connection issues
+        except S3Error as exc:
+            if exc.code == 'NoSuchKey':
+                existing_data = []
+            else:
+                raise  # Re-raise any other exceptions
+    
+        # Ensure existing data is a list (assuming the JSON data structure is a list of events)
+        if not isinstance(existing_data, list):
+            print("The existing data is not a list. Cannot append new data.")
+            return
+        
+        # Append the new event data to existing data
+        existing_data.append(self._event_json_data)
+
+        # Convert updated data back to JSON string
+        json_data = json.dumps(existing_data)
+
+        # Write the updated data back to MinIO
+        self._minio_client.put_object(
+            bucket_name=self._root_path,
+            object_name=file_path,
+            data=BytesIO(json_data.encode("utf-8")),
+            length=len(json_data),
+            content_type="application/json"
+        )
+
+        print(f"JSON file saved to MinIO at: {file_path}")
+
+
+# ==================================== #
+# (4) Set up EventConsumer
+# ==================================== #
+
+class EventConsumer:
+    def __init__(self, file_writer: FileWriter, consumer: Consumer, topics: List[str], min_commit_count: int):
+        self._file_writer = file_writer
+        self._consumer = consumer
+        self._topics = topics
+        self._min_commit_count = min_commit_count
+        
+
+    def run_consume_loop(self):
+        try:
+            self._consumer.subscribe(self._topics)
+            msg_count = 0
+            while True:
+                msg = self._consumer.poll(timeout=1.0)
```

Comment:
- switch to logging
 - think of how to consume and write multiple messages in a batch - it's more real-life-like scenario, to not do too many wirtes (on each event)